corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js

```diff
@@ -171,6 +171,163 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
         },
     };
 
+    ko.bindingHandlers.multi_sortable = {
+        updateSortableList: function(itemList) {
+            _(itemList()).each(function(item, index) {
+                if (item._sortableOrder === undefined) {
+                    item._sortableOrder = ko.observable(index);
+                } else {
+                    item._sortableOrder(index);
+                }
+            });
+        },
+        getList: function(valueAccessor) {
+            /* this function's logic follows that of ko.bindingHandlers.foreach.makeTemplateValueAccessor */
+            var modelValue = valueAccessor(),
+                unwrappedValue = ko.utils.peekObservable(modelValue);
+            if ((!unwrappedValue) || typeof unwrappedValue.length === "number") {
+                return modelValue;
+            } else {
+                return unwrappedValue['data'];
+            }
+        },
+        init: function(element, valueAccessor, allBindingsAccessor, viewModel, bindingContext) {
+            var list = ko.bindingHandlers.multi_sortable.getList(valueAccessor);
+            var forceUpdate = function() {
+                ko.bindingHandlers.multi_sortable.update(
+                    element, valueAccessor, allBindingsAccessor, viewModel, bindingContext
+                );
+            };
+            list.subscribe(forceUpdate);
+
+            // based on https://jsfiddle.net/hQnWG/614/
```

Comment:
A little bit of actual commenting would help; this is a long bit of js to comprehend.

Reply:
I think this handler might undergo significant changes before the end of the quarter, so I'm going to hold off on commenting for an audience who isn't me until it's more finalized.  If by the second week of July I haven't done so, I welcome you to hold me accountable :)

Reply:
Makes sense, will do ;)

Reply:
https://github.com/dimagi/commcare-hq/pull/21299